share-coordinator/src/main/java/org/apache/kafka/coordinator/share/ShareCoordinatorShard.java

```diff
@@ -212,6 +210,7 @@ private void handleShareSnapshot(ShareSnapshotKey key, ShareSnapshotValue value)
         shareStateMap.put(mapKey, offsetRecord);
         // if number of share updates is exceeded, then reset it
         if (snapshotUpdateCount.containsKey(mapKey)) {
+            snapshotUpdateCount.putIfAbsent(mapKey, 0); // handle mapping to null
```

Comment:
Hmm, why do we need this line? The previous line tests that the key exists.

Reply:
added in response to [contiansKey](https://github.com/apache/kafka/pull/17149#discussion_r1783239363)

Reply:
Added due to miscommunication in previous review. Will fix